src/libratbag.h

```diff
@@ -416,6 +418,23 @@ ratbag_device_new_from_udev_device(struct ratbag *ratbag,
 				   struct ratbag_device **device);
 
 /**
+ * @ingroup base
+ *
+ * Create a new ratbag context from the given path.
+ *
+ * The device is refcounted with an initial value of at least 1.
+ * Use ratbag_device_unref() to release the device.
+ *
+ * @param ratbag A previously initialized ratbag context
+ * @param path The path of the device (either /dev/input/eventX or the sysfs path)
+ *
+ * @return a new device based on the device path.
+ */
+struct ratbag_device *
+ratbag_device_new_from_path(struct ratbag *ratbag,
+			    const char *path);
+
```

Comment:
do we have a use-case for the sysfs path? I'd rather have two functions then, `new_from_device_node` and `new_from_sysfs`. Otherwise we're opening ourselves up to guesswork and behavioural dependencies.

Not as much of a problem now that libratbag is internal, but we might as well get that right from the start.

Reply:
I don't think anybody sane will use the sysfs path. But As mentioned earlier, it was just duplicating code (thus RFC).
It'll be only used by the python wrapper, which is not installed/exported, so I guess we can control what uses it or not.